api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/aup/AcceptableUsagePolicyProperties.java

```diff
@@ -64,6 +64,32 @@
     @RequiredProperty
     private String aupAttributeName = "aupAccepted";
 
+    /**
+     * Scope options for the default aup repository can store flag indicating acceptance.
+     * Scope refers to duration that acceptance is kept.
+     * Current options are global on the particular server (not replicated across CAS servers)
+     * and once per authentication via credentials (not authentication events via TGT).
+     */
+    public enum DefaultAupScope {
+        /**
+         * Store in global map for life of server.
+         */
+        GLOBAL_SERVER,
+
+        /**
+         * Store in flow scope and flow only reaches aup when user authenticates, not when they login via TGT.
```

Comment:
It might be better to explain what the value does, rather than discuss implementation details. A module should be free to process and honor this value as it wants, where the explanation should not indicate concrete choices. This sort of thing is rather hard to maintain and keep accurate, especially if for instance, the flow scope, later on, got swapped out by some other flow. Something simpler might do such as:

> Store the accepted usage policy choice for the length of the SSO session, and do not prompt for acceptance again, unless the authentication is renewed or SSO is lost.